networkx/convert_matrix.py

```diff
@@ -1127,7 +1136,9 @@ def from_numpy_array(A, parallel_edges=False, create_using=None):
                 {
                     name: kind_to_python_type[dtype.kind](val)
                     for (_, dtype, name), val in zip(fields, A[u, v])
-                },
+                }
+                if edge_attr
```

Comment:
This `if` statement will result in an empty dict for many different values that evaluate to `False`. The docs mention False and None, but others like `0`, `""` and any empty container would act like `None`.  Would it be better to check e.g. `if edge_attr is None`?

Reply:
That's a good point — I'll open a new PR as this one is merged!